core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -700,6 +700,7 @@ protected Scope visitDelete(Delete node, Optional<Scope> scope)
             node.getWhere().ifPresent(where -> analyzeWhere(node, tableScope, where));
 
             analysis.setUpdateType("DELETE", tableName, Optional.of(table), Optional.empty());
+            analysis.setTableToBeUpdated(table);
```

Comment:
this, and the other change in StatementAnalyzer, doesn't belong to "Extract table to be updated/deleted as a separate variable" commit, does it?

Reply:
this should be part of "Ensure delete/update is applied on actual table specified in query"

Reply:
I guess it belongs to the same commit. It is where the new API is being used.